### PR TITLE
DOCSP-40805 Indexes landing page

### PR DESCRIPTION
**Pull request by jordan-smith721**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-NNNNN>
Staging - <https://docs-mongodbcom-staging.corp.mongodb.com/drivers/docsworker-xlarge/NNNNN/>

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Did you run a spell-check?
- [ ] Did you run a grammar-check?
- [ ] Are all the links working?
- [ ] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/indexes.txt [hunk 1]
```diff
@@ -22,4 +22,73 @@ Optimize Queries with Indexes
...
 
-   /work-with-indexes
\ No newline at end of file
+   /work-with-indexes
+
+Overview
+--------
+
+On this page, you can see copyable code examples that demonstrate how to work with common
+types of indexes that you can use with the {+driver-short+}.
```
Comment (norareidy) [inline source/indexes.txt:31]: S: just to make this sentence a little less wordy:

```suggestion
On this page, you can see copyable code examples that demonstrate how to use the {+driver-short+} to work with common
types of indexes.
```

File: source/indexes.txt [hunk 2]
```diff
@@ -22,4 +22,73 @@ Optimize Queries with Indexes
...
+The following code shows how to create an ascending single-field index:
+
+.. literalinclude:: /includes/usage-examples/index-code-examples.cpp
+   :start-after: start-single-field
+   :end-before: end-single-field
+   :language: cpp
+   :copyable:
+
+To learn more about single-field indexes, see the
+:ref:`cpp-single-field-index` guide.
```
Comment (norareidy) [inline source/indexes.txt:66]: S: since this isn't a full guide, just a section of the "Work with Indexes" guide, I'd reword to something like:
```suggestion
:ref:`cpp-single-field-index` section of the Work with Indexes guide.
```

Comment (norareidy) [reply source/indexes.txt:66]: Applies to all

**Review by norareidy — APPROVED** [review]
LGTM with a couple suggestions!

File: source/includes/usage-examples/index-code-examples.cpp [hunk 3]
```diff
@@ -0,0 +1,18 @@
+// start-single-field
+auto index_specification = make_document(kvp("<field name>", 1));
```
Comment (kevinAlbs) [inline source/includes/usage-examples/index-code-examples.cpp:2]:
```suggestion
auto index_specification = make_document(kvp("title", 1));
```

Suggest using "title" to create index ("title_1") referenced in subsequent examples.

File: source/includes/usage-examples/index-code-examples.cpp [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+std::string result = collection.create_index(std::move(index_specification));
+
+std::cout << "Index created: " << result << std::endl;
```
Comment (kevinAlbs) [inline source/includes/usage-examples/index-code-examples.cpp:5]:
```suggestion
auto result = collection.create_index(index_specification.view());
std::cout << "Index created: " << bsoncxx::to_json(result) << std::endl;
```

`create_index` returns a `bsoncxx::document::value`, not a `std::string`. Suggest using `auto` and `bsoncxx::to_json(result)` to fix compile.

Additionally, suggest using `.view()` rather than `std::move` to avoid unnecessary ownership transfer.

**Review by kevinAlbs — CHANGES_REQUESTED** [review]

File: source/includes/usage-examples/index-code-examples.cpp (hunk 3 above)
Comment (jordan-smith721) [reply source/includes/usage-examples/index-code-examples.cpp:2]: By convention we use placeholders on these landing pages (formerly known as "usage example" pages) instead of giving actual fields so that the user can copy/paste these and easily see where to swap out for their specific use case

Comment (kevinAlbs) [reply source/includes/usage-examples/index-code-examples.cpp:2]: Got it. In that case, consider updating the example below to replace `title_1` with `<field name>_1`.

**Review by kevinAlbs — APPROVED** [review]
LGTM with possible replacement of `title_1`.

File: source/includes/usage-examples/index-code-examples.cpp (hunk 3 above)
Comment (jordan-smith721) [reply source/includes/usage-examples/index-code-examples.cpp:2]: Oops. Good catch, thanks!
